packages/jest-snapshot/src/Snapshot.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+'use strict';
+
+const fs = require('fs');
+
+class Snapshot {
+
+  constructor(filename) {
+    this._filename = filename;
+    this._loaded = null;
+  }
+
+  _getLoaded() {
```

Comment:
what do you mean with "the test needs to be refreshed"? The way I see it there should be at most one read operation and 0-1 write operations for snapshots. You should keep the new value in the buffer and write it at the end of the test if a mutation occurred.